### ISSUE TITLE
mattermost does not find URLs by fqdn

### ISSUE THREAD

**Issue by wwalker** [mattermost/mattermost-server#12150]
text ~ "fqdn" and text ~ "search" order by created DESC

#### Summary
Searching for a fqdn that occurs in a URL does not show up in the results.  searching without the .com does return the results.  MySQL database

#### Steps to reproduce
Create a message with a URL in it:

"The meeting is in https://cisco.webex.com/cisco/j.php?"

search for cisco.webex.com

#### Expected behavior
The message would show up in the search results.

#### Observed behavior (that appears unintentional) 
The message is not found.

#### Possible fixes
A work around is to only search for cisco.webex finds the message; however, that is very non-intuitive.
![mattermost-url-search](https://user-images.githubusercontent.com/23478/64642597-58ed8780-d3d4-11e9-9423-737fa34dd498.png)

**Comment by wwalker** [comment]
The issue isn't really around fqdn.  If I search for : webex com
You get No results.  The point is that it is okay if search ignores a word because it is too common.  But, this isn't ignoring "com", it makes ANY query have No results.

I have a message that contains "how to reorg stuff so " .

If I search for reorg I get 2 message results.

If I search for :  reorg so

I get No results

So, it appears that short words are ignored during the indexing, but Not ignored during the search.

**Comment by amyblais** [comment]
Hi @wwalker, Would you be open to reviewing this page and sending us the relevant information about your environment: https://docs.mattermost.com/process/support.html#general-questions-for-any-issues?

**Comment by wwalker** [comment]
TL;DR - installed native on ubuntu 16.04.6 with mysql 5.7.24-0ubuntu0.16.04.1
No problems with the server.  Just an issue with how the search engine works.

What OS and version is the Mattermost server installed?
Ubuntu 16.04.6 LTS

What is your Mattermost server version?
Version: 5.10.0
Build Number: 5.10.0
Build Date: Tue Apr 16 23:13:55 UTC 2019
Build Hash: dc70ab42034da17eaf23be5a86abae81b3f1cd21
Build Enterprise Ready: true
DB Version: 5.10.0

Are you experiencing the issues with the browser webapp, if so which one?
Yes, Chrome Version 77.0.3865.75 (Official Build) (64-bit)

Are you experiencing the issues with the Mattermost Desktop App, if so what version and OS?
Samr problem in the linux client.  This is a search database issue.

Can you send a snippet of the Mattermost server logs around the time of the incident?
{"level":"error","ts":1568422061.7929068,"caller":"http/server.go:1794","msg":"http: TLS handshake error from 128.107.241.164:41419: EOF","source":"httpserver"}
{"level":"info","ts":1568422074.9973319,"caller":"scheduler/worker.go:78","msg":"Worker: Job is complete","worker":"Plugins","job_id":"sm7hraq9ytyyjptzbody9yi9da"}
{"level":"info","ts":1568422075.112113,"caller":"mlog/log.go:164","msg":"Found users with LDAP configured","workername":"EnterpriseLdapSync","num_ldap_users":118}
{"level":"error","ts":1568422130.4720786,"caller":"http/server.go:1794","msg":"http: TLS handshake error from 128.107.241.177:51533: read tcp 10.7.6.60:443->128.107.241.177:51533: i/o timeout","source":"httpserver"}
{"level":"info","ts":1568422135.4063797,"caller":"scheduler/worker.go:78","msg":"Worker: Job is complete","worker":"Plugins","job_id":"i3sdnb4fupr9pf37b5hy5szjte"}
{"level":"info","ts":1568422195.8229651,"caller":"scheduler/worker.go:78","msg":"Worker: Job is complete","worker":"Plugins","job_id":"dy88ozjjj7n88n8so41ub3cbow"}
{"level":"info","ts":1568422256.2419906,"caller":"scheduler/worker.go:78","msg":"Worker: Job is complete","worker":"Plugins","job_id":"tjhijunhn78mifotkfdo3en8pc"}


Are you running Mattermost in a container and/or using container orchestration?
on ubuntu 16 native

Mattermost Is Not Working / The Server Keeps Dying
NA

Check the status of the server process
NA

sudo systemctl status mattermost.service
NA
sudo systemctl enable mattermost.service
sudo systemctl start mattermost.service
Mattermost is not running after a reboot
NA

sudo systemctl enable mattermost.service
sudo systemctl start mattermost.service
NA

Check the system logs
Not dying
sudo journalctl -u mattermost.service
NA
Database Issues
Questions to ask when providing support for database issues.

IMPORTANT: Before asking for any database queries to be run, suggest a back-up is completed to prevent accidental data loss or corruption.

PostgreSQL or MySQL?
MySQL

If they’re using Amazon Aurora MySQL, and their issue seems like a race condition, it might be due to the multi-region lag that Aurora can have. If this is the case there may be a race in code we need to fix. @christopher will know more.
Not on Aurora

What is the database version?
mysql-server-5.7                      5.7.24-0ubuntu0.16.04.1
Read replicas?
No
Connection errors?
No
Other
Nothing comes to mind
WebSocket Issues
No websocket issues

Can you send us your Nginx configuration files?
Not a web server problem.

Can you send us a snippet of your Nginx error logs around the time of the incident?
NA

LDAP Issues
NA

What AD/LDAP server and version are you using?
NA
Are there any LDAP errors in the logs?
NA

Can you send an example user from your system?
NA

Is the server properly licensed?

Edition:Mattermost Enterprise Edition. Enterprise features on this server have been unlocked with a license key and a valid subscription.
License:
This software is offered under a commercial license. See ENTERPRISE-EDITION-LICENSE.txt in your root install directory for details. See NOTICE.txt for information about open source software used in this system. Your subscription details are as follows:

Name: Chad Rodrigue
Company or organization name: Cisco
Edition: Mattermost Enterprise Edition E10
Number of users: 200
License issued: March 22, 2019 17:02 UTC
Start date of license: March 22, 2019
Expiry date of license: March 21, 2020

TLS/SSL Issues
NA
No gitlab problems

No login problems

Other Mattermost resources (i18n, email templates, webapp code, etc) are located in /opt/gitlab/embedded/service/mattermost.

NA

**Comment by amyblais** [comment]
@wwalker Thank you for providing all this information! I wasn't able to reproduce this on a v5.15 server at community.mattermost.com. I'm wondering if you're open to testing on the community server I linked or if you're open to upgrading to the latest server version to see if the issue gets resolved. Let me know your thoughts on this,

**Comment by wwalker** [comment]
I've tested on the community.mattermost server and it works fine there.  Thank you for your help.

I'll get us upgraded.
